plugins/woocommerce-admin/client/marketplace/components/header-account/header-account-modal.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import { Button, ButtonGroup, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+
+export interface HeaderAccountModalProps {
+	setIsModalOpen: ( value: boolean ) => void;
+	disconnectURL: string;
+}
+
+export default function HeaderAccountModal(
+	props: HeaderAccountModalProps
+): JSX.Element {
+	const { setIsModalOpen, disconnectURL } = props;
+	const [ isBusy, setIsBusy ] = useState( false );
+	const toggleIsBusy = () => setIsBusy( ! isBusy );
+	const closeModal = () => setIsModalOpen( false );
+
+	return (
+		<Modal
+			title={ __( 'Are you sure?', 'woocommerce' ) }
+			onRequestClose={ closeModal }
+			focusOnMount={ true }
+			className="woocommerce-marketplace__header-account-modal"
+			style={ { borderRadius: 4 } }
+			overlayClassName="woocommerce-marketplace__header-account-modal-overlay"
+		>
+			<p className="woocommerce-marketplace__header-account-modal-text">
+				{ __(
+					'Keep your your account connected to manage your subscriptions, get updates and support for your extensions and themes.',
+					'woocommerce'
+				) }
+			</p>
+			<ButtonGroup className="woocommerce-marketplace__header-account-modal-button-group">
```

Comment:
Woah, now I'm sorry I pushed for these BEM classnames! In future we can rethink what we're doing – treating the whole page as the block in BEM leads to this kind of name.